### DISCUSSION TITLE
New ADC Interface for Continuous Sampling

### DISCUSSION THREAD

**Discussion by pdietl** [zephyrproject-rtos/zephyr#48478] [Ideas]
@anangl @erwango @henrikbrixandersen @yonsch 

See this issue https://github.com/zephyrproject-rtos/zephyr/issues/39640
See this branch https://github.com/worldcoin/zephyr/tree/stm32_adc_dma

I would like to see a change to the public API of ADC to include some way to do continuous sampling. I intend this to be implemented with IRQs (if slow continuous sampling is what you want) or with DMA. I see this interface taking a buffer and calling some callback function when the provided buffer is 1/2 full.

I see three ways to go about this.

1. Create a new function for this purpose. This is what I went with here https://github.com/zephyrproject-rtos/zephyr/compare/main...worldcoin:zephyr:stm32_adc_dma#diff-0fb1cb925599ca94ff475e7ac8f4a8bf67fca7d4b8760f9a398a03188af07556R6
2. Make an addition to [`adc_sequence_options`](https://docs.zephyrproject.org/latest/hardware/peripherals/adc.html#c.adc_sequence.options) to accept a callback function, which means that one wants to use continuous mode.
3. Something else entirely, maybe using the RTIO API?

Please give your thoughts here. Thanks!

**Comment by pdietl** [comment]
Pinging @anangl again.

**Reply by erwango** [reply]
Not sure if people get notifications from in discussions. You can try to ping him in discord

**Reply by pdietl** [reply]
@erwango what is his Discord handle?

**Reply by erwango** [reply]
not sure, but I guess he should be monitoring ADC channel

**Reply by erwango** [reply]
^^@anangl Did you reach to a conclusion on that matter ?
